corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -1,38 +1,71 @@
+        "data_interfaces/js/case_rule_criteria",
+        "scheduling/js/create_schedule.ko",
+        "data_interfaces/js/make_read_only",
+    ],
+    function ($, ko, _, initialPageData) {
+        function basicInformationTab(name) {
+            var self = {};
+            self.name = ko.observable(name);
+            self.basicTabValid = ko.computed(function () {
+                return !_.isEmpty(self.name().trim());
+            });
+            self.setRuleTabVisibility = function () {
+                if (self.basicTabValid()) {
+                    $("#rule-nav").removeClass("hidden");
+                }
+            };
+            self.navigateToNav = function (navId) {
+                $(navId).find("a").trigger("click");
+            };
+            self.handleBasicNavContinue = function () {
                 $("#rule-nav").removeClass("hidden");
-            }
-        };
-        self.navigateToNav = function (navId) {
-            $(navId).find('a').trigger('click');
-        };
-        self.handleBasicNavContinue = function () {
-            $("#rule-nav").removeClass("hidden");
-            $('#rule-nav').find('a').trigger('click');
-        };
-        return self;
-    }
+                $("#rule-nav").find("a").trigger("click");
+            };
+            return self;
+        }
+
+        function getFormContents() {
+            return _.object(
+                _.map($(".main-form").serializeArray(), function (obj) {
+                    return [obj.name, obj.value];
+                })
+            );
+        }
+
+        $(function () {
+            var name = initialPageData.get("rule_name");
+            var basicInformation = basicInformationTab(name);
+            // setup tab
+            basicInformation.setRuleTabVisibility();
+            $("#conditional-alert-basic-info-panel").koApplyBindings(
+                basicInformation
+            );
 
-    $(function () {
-        var name = initialPageData.get('rule_name');
-        var basicInformation = basicInformationTab(name);
-        // setup tab
-        basicInformation.setRuleTabVisibility();
-        $("#conditional-alert-basic-info-panel").koApplyBindings(basicInformation);
-    });
-});
+            var initialFormContent = getFormContents();
+
+            $(".main-form").on("submit", function () {
+                var finalFormContent = getFormContents();
+                // compare changes with initialFormContent to check if only 'message' has changed
+                var formContentChanges = [];
+                Object.keys(finalFormContent).forEach(function (key) {
+                    if (finalFormContent[key] !== initialFormContent[key]) {
+                        formContentChanges.push(key);
+                    }
+                });
+                if (!_.isEqual(["content-message"],formContentChanges)) {
+                    // to convey to HQ to reprocess or not
+                    $("#initiate_rule").val(false);
```

Comment:
nit: is there a way we can handle this in the loop itself?
As soon as we find a key that is not in ['content-message'], we know that the rule needs to be initiated and can just set it to true.

Reply:
I think we will need to change `forEach` to `for` or `for` loop because `forEach` can't be break' ed in between.
In terms of readability, I am not sure which approach would be good.
@mkangia what would be your comments?

Reply:
[_.find](https://underscorejs.org/#find) can be a readable way to set up a loop that breaks on a condition.